frontends/web/src/contexts/DarkmodeProvider.tsx

```diff
@@ -0,0 +1,92 @@
+/**
+ * Copyright 2023 Shift Crypto AG
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { useState, useEffect, ReactNode, useCallback } from 'react';
+import { getConfig, setConfig } from '../api/backend';
+import { setDarkTheme } from '../api/darktheme';
+import { useMediaQuery } from '../hooks/mediaquery';
+import DarkModeContext from './DarkmodeContext';
+
+type TProps = {
+  children: ReactNode;
+}
+
+export const DarkModeProvider = ({ children }: TProps) => {
+  const [isDarkMode, setIsDarkMode] = useState(false);
+  const osPrefersDarkmode = useMediaQuery('(prefers-color-scheme: dark)');
+
+  const setAppTheme = useCallback(() => {
+    setDarkTheme(isDarkMode);
+    if (isDarkMode) {
+      document.body.classList.add('dark-mode');
+      document.body.classList.remove('light-mode');
+    } else {
+      document.body.classList.remove('dark-mode');
+      document.body.classList.add('light-mode');
+    }
+  }, [isDarkMode]);
```

Comment:
Here we used the new `setDarkTheme` created in #2035 . CCing @Beerosagos and @thisconnect in case wanting to take a look once more before I merge.

Reply:
also wrapped it in a function `setAppTheme` just so it's neater:)